src/ray/common/task/task_spec.cc

```diff
@@ -126,7 +126,7 @@ bool TaskSpecification::HasRuntimeEnv() const {
 }
 
 int TaskSpecification::GetRuntimeEnvHash() const {
-  WorkerCacheKey env = {OverrideEnvironmentVariables(), SerializedRuntimeEnv()};
+  WorkerCacheKey env = {OverrideEnvironmentVariables(), SerializedRuntimeEnv(), "{}"};
```

Comment:
Why it's empty for resources?

Reply:
The `TaskSpecification::GetRuntimeEnvHash` will be used in `SchedulingKey` which don't have allocation resource.

Reply:
I have refact it and add  `required_resources` in WorkerCacheKey